node/silkworm/downloader/internals/chain_elements.hpp

```diff
@@ -78,7 +78,7 @@ struct Link {
 
     void remove_child(std::shared_ptr<Link> child) {
```

Comment:
It might not be an issue here, but it is not recommended to take a std::shared_ptr as a value, because every call causes a reference increment/decrement. Better take a `const std::shared_ptr<Link>&`, or even a `const Link&`

Reply:
Yes, I agree. `child` doesn't leak so there's not need for `shared_ptr` here, `const &` should be just fine.

Reply:
ok, I will do it in the next commit

Reply:
done